### PR TITLE
backup: import macros with the use syntax

### PR DESCRIPTION
**Pull request by skyline75489**
Signed-off-by: skyline75489 <skyline75489@outlook.com>

### What problem does this PR solve?

Issue Number: supports #9678 

Problem Summary:

### What is changed and how it works?

Import macros from other crates with the `use` syntax.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- No release note

### CODE REVIEW

File: components/backup/src/lib.rs [hunk 1]
```diff
@@ -3,12 +3,8 @@
 #![recursion_limit = "200"]
 #![feature(box_patterns)]
 
-#[macro_use]
-extern crate failure;
 #[allow(unused_extern_crates)]
 extern crate tikv_alloc;
```
Comment (skyline75489) [inline components/backup/src/lib.rs:7]: Is this really used? Not sure If I can just remove it.

Comment (JmPotato) [reply components/backup/src/lib.rs:7]: You can try to remove it and build again to see whether it's necessary.

**Comment by sticnarf** [comment]
I am not so sure about this change. `#[macro_use]` simplifies the uses of these macros if it doesn't cause any confusion. Wdyt? @nrc

**Review by nolouch — APPROVED** [review]

**Review by nrc — APPROVED** [review]
lgtm

**Comment by nrc** [comment]
> I am not so sure about this change. #[macro_use] simplifies the uses of these macros if it doesn't cause any confusion. Wdyt? @nrc

I definitely prefer using `use` to import macros - it is more consistent, more flexible, and makes it easier to lookup the macro definition.

File: components/backup/src/lib.rs (hunk 1 above)
Comment (JmPotato) [reply components/backup/src/lib.rs:7]: Any update on this problem?

Comment (skyline75489) [reply components/backup/src/lib.rs:7]: Well I found this https://github.com/tikv/tikv/blob/2a2fa03da53b63f3fc24d7ea53aead40176979b5/components/tikv_alloc/src/lib.rs#L10-L12

but I'm not sure this is still true

Comment (JmPotato) [reply components/backup/src/lib.rs:7]: Well, I think we better leave it unchanged.

**Pull request merged**